langlib/lang.int/src/main/java/org/ballerinalang/langlib/integer/Step.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.langlib.integer;
+
+import io.ballerina.runtime.api.creators.ErrorCreator;
+import io.ballerina.runtime.internal.util.exceptions.BLangExceptionHelper;
+import io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons;
+import io.ballerina.runtime.internal.util.exceptions.RuntimeErrors;
+
+import static io.ballerina.runtime.api.constants.RuntimeConstants.INT_LANG_LIB;
+import static io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons.getModulePrefixedReason;
+
+/**
+ * Native implementation of lang.int:externalCheckStep(int).
+ *
+ * @since 2201.3.0
+ */
+public class Step {
+
+    public static long checkStep(long step) {
+        if (step == 0) {
+            throw ErrorCreator.createError(getModulePrefixedReason(INT_LANG_LIB,
```

Comment:
is this the reason, we can't write this logic in ballerina?

Reply:
@MaryamZi @gimantha We need to find a solution for this.

Reply:
Raised the same in https://github.com/ballerina-platform/ballerina-lang/pull/38279#discussion_r1010331305. I believe we should be able to write this in b7a itself, it's just a panic?

Reply:
I was not able to create the panic in ballerina, I'll check and update.

Reply:
When trying to use `panic` it throws a `NullPointerException` at 
`org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolResolver.lookupMethodInModule(SymbolResolver.java:942)`
for `moduleSymbol`.
`langMapModuleSymbol` might not be still initialized at this point and cause this.

Reply:
The map langlib doesn't use the int langlib (or any other langlib), right? So we may able to get this to work by loading the map langlib before the int langlib. 

io.ballerina.projects.Bootstrap#loadLangLibSymbols
io.ballerina.projects.Bootstrap#loadLangLib

Reply:
Created new issue for this https://github.com/ballerina-platform/ballerina-lang/issues/38608
As discussed, we'll keep the external function for this PR.